src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Android.cs

```diff
@@ -0,0 +1,180 @@
+
+                if (ciphertextBytesWritten != ciphertextAndTag.Length)
+                {
+                    Debug.Fail($"GCM encrypt wrote {ciphertextBytesWritten} of {ciphertextAndTag.Length} bytes.");
+                    throw new CryptographicException();
+                }
+
+                ciphertextAndTag[..ciphertext.Length].CopyTo(ciphertext);
+                ciphertextAndTag[ciphertext.Length..].CopyTo(tag);
+            }
+            finally
+            {
+                if (rented != null)
+                {
+                    CryptoPool.Return(rented, ciphertext.Length + tag.Length);
+                }
+            }
+        }
+
+        private void DecryptInternal(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> tag,
+            Span<byte> plaintext,
+            ReadOnlySpan<byte> associatedData)
+        {
+            if (!Interop.Crypto.EvpCipherSetTagLength(_ctxHandle, tag.Length))
+            {
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+
+            Interop.Crypto.EvpCipherSetKeyAndIV(
+                _ctxHandle,
+                ReadOnlySpan<byte>.Empty,
+                nonce,
+                Interop.Crypto.EvpCipherDirection.Decrypt);
+
+            if (associatedData.Length != 0)
+            {
+                if (!Interop.Crypto.EvpCipherUpdateAAD(_ctxHandle, associatedData))
+                {
+                    throw Interop.Crypto.CreateOpenSslCryptographicException();
+                }
+            }
+
+            if (!Interop.Crypto.EvpCipherUpdate(_ctxHandle, plaintext, out int plaintextBytesWritten, ciphertext))
+            {
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+
+            if (!Interop.Crypto.EvpCipherUpdate(_ctxHandle, plaintext.Slice(plaintextBytesWritten), out int bytesWritten, tag))
+            {
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+
+            plaintextBytesWritten += bytesWritten;
+
+            if (!Interop.Crypto.EvpCipherFinalEx(
+                _ctxHandle,
+                plaintext.Slice(plaintextBytesWritten),
+                out bytesWritten))
+            {
+                CryptographicOperations.ZeroMemory(plaintext);
+                throw new CryptographicException(SR.Cryptography_AuthTagMismatch);
+            }
+
+            plaintextBytesWritten += bytesWritten;
+
+            if (plaintextBytesWritten != plaintext.Length)
+            {
+                Debug.Fail($"GCM decrypt wrote {plaintextBytesWritten} of {plaintext.Length} bytes.");
+                throw new CryptographicException();
+            }
+        }
+
+        private static IntPtr GetCipher(int keySizeInBits)
+        {
+            switch (keySizeInBits)
```

Comment:
Still learning all the new C# syntax, but this could also be converted to the following, right?

```C#
return keySizeInBits switch
{
    ...
    _ => IntPtr.Zero;
};
```